### PR TITLE
Clent-side flatline interpreters (JS and Python)

### PR DESCRIPTION
**Pull request by jaor**
@mmerce could you please give it a look? thanks!

### CODE REVIEW

File: python/flatline/interpreter.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        return self._context.call('f__.flatline.%s' % fun, *args)
+
+    @staticmethod
+    def infer_fields(row):
+        """Utility function generating a mock list of fields.
+
+        Usually, checks and applications of Flatline expressions run
+        in the context of a given dataset's field descriptors, but
+        during testing it's useful sometimes to provide a mock set of
+        them, based on the types of the values of the the test input
```
Comment (mmerce) [inline python/flatline/interpreter.py]: typo: the the

File: python/flatline/interpreter.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        return dataset
+
+    def defined_functions(self):
+        """A list of the names of all defined Flaline functions"""
+        return self.__eval_in_flatline('defined_primitives')
+
+    def check_lisp(self, sexp, dataset={}):
+        """Checks whether the given lisp s-expression is valid.
+
+        Any operations referring to a dataset fields will use the
```
Comment (mmerce) [inline python/flatline/interpreter.py]: typo: to dataset fields

File: python/flatline/interpreter.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    def __dataset(dataset, rows):
+        if dataset is None and len(rows) > 0:
+            return {'fields': Interpreter.infer_fields(rows[0])}
+        return dataset
+
+    def defined_functions(self):
+        """A list of the names of all defined Flaline functions"""
+        return self.__eval_in_flatline('defined_primitives')
+
+    def check_lisp(self, sexp, dataset={}):
```
Comment (mmerce) [inline python/flatline/interpreter.py]: It's not good practice to have dicts or arrays as default values in arguments. They can be modified in the inner code of the function and remain changed, but you will know better if that's the case.

File: python/flatline/interpreter.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        information found in the provided dataset, which should have
+        the structure of the 'object' component of a BigML dataset
+        resource.
+
+        """
+        r = self.__eval_in_flatline('evaluate_sexp', sexp, dataset)
+        r.pop(u'mapper', None)
+        return r
+
+    def check_json(self, json_sexp, dataset={}):
```
Comment (mmerce) [inline python/flatline/interpreter.py]: ditto about dataset arg

File: python/flatline/sampler.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        """Returns a list of lists representing the current sample's rows.
+
+        See 'take_sample' for updating the current sample and 'sample'
+        for the full set of its properties.
+
+        """
+        return self.sample().get('rows')
+
+    def fields(self):
+        """The list of field descriptiors for the current sample.
```
Comment (mmerce) [inline python/flatline/sampler.py]: typo: descriptors

File: python/flatline/interpreter.py (hunk 3 above)
Comment (jaor) [inline python/flatline/interpreter.py]: yep, i'll change that to None, which is also valid

File: readme.md [hunk 6]
```diff
@@ -23,6 +23,16 @@ dataset).
...
 
+## Local interpreters
+
+We include in [js](./js) Flatline interpreters implemented in
+Javascript (compiled by Clojurescript from our canonical server-side
+implementation) that you can use from your browser or from a nodejs
+session.
+
+The [python directory](./python) contains a small Python library that
+wraps the nodejs interpreter and let's you interact with it using
```
Comment (mmerce) [inline readme.md]: typo: lets you

**Comment by mmerce** [comment]
Looks good. Just some typos.

**Pull request merged**